tools/roslaunch/src/roslaunch/rlutil.py

```diff
@@ -207,6 +207,10 @@ def check_roslaunch(f):
         file_deps = {}
     for pkg, miss in missing.items():
         if miss:
+            print("Missing package dependencies: %s/package.xml: %s"%(pkg, ', '.join(miss)))
+        # even if the pkgs is not found in packges.xml, if other package.xml provdes that pkgs, then it will be ok                                                                                           
+        miss_all = list(set(miss) - set(reduce(lambda x, y: x+y, [file_deps[pkg].pkgs for pkg in file_deps.keys()])))
```

Comment:
The logic for the lambda looks overly complicated. Please try to simply the expression.